src/sentry/integrations/vsts_extension/integration.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import absolute_import
+
+from django.contrib import messages
+from django.http import HttpResponseRedirect
+
+from sentry.integrations.vsts.integration import VstsIntegrationProvider
+from sentry.pipeline import PipelineView
+from sentry.utils.http import absolute_uri
+
+
+class VstsExtensionIntegrationProvider(VstsIntegrationProvider):
+    key = 'vsts-extension'
+
+    def get_pipeline_views(self):
+        views = super(VstsExtensionIntegrationProvider, self).get_pipeline_views()
+        views[-1] = VstsExtensionFinishedView()
+        return views
+
+    def build_integration(self, state):
+        # Normally this is saved into the ``identity`` state, but for some
+        # reason it gets wiped out in the NestedPipeline. Instead, we'll store
+        # it in it's own key (``vsts``) to be used down the line in
```

Comment:
Can you tell me a bit more about this? it's not a blocker to merging here, but that sounds a bit odd.

Reply:
I'm not completely sure, to be honest. Whenever I stored data in the `identity` part of state, it was being cleared once the IdentityPipeline kicked in. I'm guessing it does something like `state['identity'] = {}` somewhere. It definitely is odd.

Reply:
do you mean 'account' is stored in the identity state? it shouldn't be, identity is just the user identity?

this code is pretty confusing, but it seems like https://github.com/getsentry/sentry/blob/master/src/sentry/pipeline/__init__.py#L74 is where the identity key is overwritten

Reply:
It might be different now. When I wrote that comment, it looked like the `VstsIdentityProvider` was storing `AccountName` and `AccountId` in the `identity` state.

Reply:
That's definitely not right, I'd like to know more about this. If you have a chance @mnoble could you walk me through what's going on in person?

Reply:
This is no longer the case, anyway. @adhiraj took the `AccountName` and `AccountId` out of the identity state a while ago, so this comment is outdated. I'm just going to remove the comment.